compiler/rustc_mir_transform/src/jump_threading.rs

```diff
@@ -149,11 +149,8 @@ impl Condition {
         (self.value == value) == (self.polarity == Polarity::Eq)
     }
 
-    fn inv(mut self) -> Self {
-        self.polarity = match self.polarity {
-            Polarity::Eq => Polarity::Ne,
-            Polarity::Ne => Polarity::Eq,
-        };
+    fn not(mut self, is_bool: bool) -> Self {
+        self.value = self.value.not(is_bool);
```

Comment:
I am not sure we want this as an operation on `ScalarInt`. We don't do this for any other operation. Instead, this should call `ecx.unary_op`.